### ISSUE TITLE
Get Document tree traversal

### ISSUE THREAD

**Issue by Salrandin** [graphql-java/graphql-java#1146]
Hi, I have been looking at the NodeTraverser and QueryTraversal code and was wondering if there was an easy way to all the GraphQLObjectTypes in postOrder traversal order with all its fields.  Looking at the QueryTraversal, it requires a GraphQLSchema object which is fine but when I generate it through an IDL it requires a RunTimeWiring which I do not want to provide.

Is there a way to get all your GraphQL objects in your document with its corresponding fields in a tree traversal order without providing resolvers?  Thanks.

**Comment by bbakerman** [comment]
Have  a a look at graphql.schema.idl.UnExecutableSchemaGenerator

It is an internal class (and hence may change) but shows you how to have a No op runtime wiring.

Now it technically not possible to get an EXACT list of the graphql object types because it is ONLY during the execution of a query that interfaces / unions can be turned into an actual object type.

We would like to know more about what you want to achieve in this "AST  -> graphql object types" traversal.  This will help us give you a better answer

**Comment by Salrandin** [comment]
Hi, thanks for the advice so far.

For our project we are currently leveraging https://github.com/graphql-java/graphql-java-tools.

This requires us to make pojos for every type in our schema and also data resolvers.

We are creating expanding on our current schema with a few hundred new attributes and creating these pojos and resolvers by hand is cumbersome and seems backwards.  We are trying to use java poet code generation to create these class shells and also use it to make sure our java classes are in sync with our schema as eventually several teams will be touching this code.

So in short we would like to do some schema introspection during build time to populate all the necessrary java classes.  We can do this with recursion but wanted to see if its possible to leverage Graphql-java code to do this instead.

**Comment by bbakerman** [comment]
We don't have a code construct to traverse the schema per se because its not something that graphql-java does at runtime (as opposed to say query traversal).  I think you may want to write your own.

Remember that the type system is circular so you have to remember what you have seen.

I am going to close this since I dont think we should add more code to graphql-java for this since its yet more cost to maintain but not commonly asked for.  We can revist that decision of course
